api/v1/api.proto

```diff
@@ -0,0 +1,314 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// To compile, run 'protoc -I api/v1 api/v1/api.proto --go_out=plugins=grpc:api/v1' in rkt root directory.
+
+syntax = "proto3";
+
+package api;
+
+// ImageType defines the supported image type.
+enum ImageType {
+        IMAGE_TYPE_UNDEFINED = 0;
+        IMAGE_TYPE_APPC      = 1;
+        IMAGE_TYPE_DOCKER    = 2;
+        IMAGE_TYPE_OCI       = 3;
+}
+
+// ImageFormat defines the format of the image.
+message ImageFormat {
+        // Type of the image, required.
+        ImageType type = 1;
+
+        // Version of the image format, required.
+        string version = 2;
+}
+
+// Image describes the image's information.
+message Image {
+        // The format of the image.
+        ImageFormat format = 1;
+
+        // ID of the image,  a string of the format 'hash-value'.
+        // Currently the only permitted hash algorithm is sha512.
+        string id = 2;
+
+        // Name of the image in the image manifest, e.g. 'coreos.com/etcd'.
+        string name = 3;
+
+        // Version of the image, e.g. 'latest', '2.0.10'.
+        string version = 4;
+
+        // The timestamp of when the image is imported.
+        int64 import_timestamp = 5;
+
+        // The JSON-encoded string that represents the manifest.
+        string manifest = 6;
+}
+
+// Network describes the network information of a pod.
+message Network {
+        // The name of the network that a pod belongs to.
+        string name = 1;
+
+        // The pod’s IPv4 address within the network, optional if IPv6 address is given.
+        string ipv4 = 2;
+
+        // The pod’s IPv6 address within the network, optional if IPv4 address is given.
+        string ipv6 = 3;
+}
+
+// AppState defines the possible states of the app.
+enum AppState {
+        APP_STATE_UNDEFINED = 0;
+        APP_STATE_RUNNING   = 1;
+        APP_STATE_EXITED    = 2;
+}
+
+// App describes the information of an app that's running in a pod.
+message App {
+        // Name of the app.
+        string name = 1;
+
+        // The image used by the app. Note that it may only contain the image id.
+        Image image = 2;
+
+        // The state of the app. Undefined if the app is not returned by InspectPod().
+        AppState state = 3;
+
+        // The exit code of the app.
+        // Undefined if the app is not returned by InspectPod(), or the app has not exited.
+        sint32 exit_code = 4;
+}
+
+// PodState defines the possible states of the pod.
+enum PodState {
+        POD_STATE_UNDEFINED       = 0;
+        POD_STATE_EMBRYO          = 1;
```

Comment:
Can you document the semantics of each of these states? I assume the kubelet will have to directly deal with these states.